src/main/java/pl/edu/uj/engine/eventloop/EventLoopThread.java

```diff
@@ -114,15 +114,17 @@ public void run() {
                 }
             }
 
-            if (callbackStorage.isEmpty() && eventLoopQueue.isEmpty()) {
+            if (eventLoopThreadRegistry.returnEventLoopThread(jar) == 0 && callbackStorage.isEmpty() && eventLoopQueue.isEmpty()) {
                 logger.info("No more callbacks to execute, shutting down");
-                eventPublisher.publishEvent(new JarJobsCompletedEvent(this, getJar()));
-                break;
+                if (eventLoopThreadRegistry.unregister(jar).isPresent()) {
```

Comment:
if other thread hold reference to this elt then it will stop and it will stay at registry what is an error